internal/services/network/subnet_resource.go

```diff
@@ -320,14 +392,36 @@ func resourceSubnetUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 		props.Delegations = expandSubnetDelegation(delegationsRaw)
 	}
 
-	if d.HasChange("enforce_private_link_endpoint_network_policies") {
-		v := d.Get("enforce_private_link_endpoint_network_policies").(bool)
-		props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(v))
-	}
+	if features.FourPointOhBeta() {
+		if d.HasChange("private_endpoint_network_policies_enabled") {
+			v := d.Get("private_endpoint_network_policies_enabled").(bool)
+			props.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetNetworkPolicy(v))
+		}
 
-	if d.HasChange("enforce_private_link_service_network_policies") {
-		v := d.Get("enforce_private_link_service_network_policies").(bool)
-		props.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(v))
+		if d.HasChange("private_link_service_network_policies_enabled") {
+			v := d.Get("private_link_service_network_policies_enabled").(bool)
+			props.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetNetworkPolicy(v))
+		}
+	} else {
+		if d.HasChange("enforce_private_link_endpoint_network_policies") || d.HasChange("private_endpoint_network_policies_enabled") {
+			if d.HasChange("enforce_private_link_endpoint_network_policies") {
```

Comment:
I think if a user switches which of these properties they are using in config, this will potentially result in unexpected/incorrect behaviour. I suspect we'll need to process both values and give priority to the changes on the newly introduced property to ensure that users migrating to that will get their expected outcome?

Reply:
Yeah this is the tricky bit because you have no idea if it is set or not... still trying to work this bit out... 🤔